doc/getting-started/rt_industry.rst

```diff
@@ -0,0 +1,390 @@
+   ACRN:\>cpuid 0x10 0x2
+   cpuid leaf: 0x10, subleaf: 0x2, 0x7:0x0:0x0:0x3
+
+Set CLOS (QOS MASK) and PQR_ASSOC MSRs to configure the CAT
+```````````````````````````````````````````````````````````
+ApolloLake doesn't have L3 cache, and supports L2 CAT. The CLOS MSRs are per L2 cache,
+starts from 0x00000D10, in the case there is 4 CLOS MSRs, the address of CLOS MSRs::
+
+   MSR_IA32_L2_QOS_MASK_0    0x00000D10
+   MSR_IA32_L2_QOS_MASK_1    0x00000D11
+   MSR_IA32_L2_QOS_MASK_2    0x00000D12
+   MSR_IA32_L2_QOS_MASK_3    0x00000D13
+
+And the PQR_ASSOC MSR is per CPU core, each core has its own PQR_ASSOC::
+
+   MSR_IA32_PQR_ASSOC        0x00000C8F
+
+To set the CAT, we need to set the CLOS MSRs and then set PQR_ASSOC of each CPU,
+so that the CPU of RTVM to use dedicated cache ways and other CPUs to use the cache ways.
+Taking a Quad Core ApolloLake platform for example, CPU0 and CPU1 share a L2 cache
+and CPU2 and CPU3 share the other L2 cache.
+
+- If we allocate CPU2 and CPU3, there is no extra action required.
+- If we allocate only CPU1 to the RTVM, we need to set CAT as follows.
+  These commands actually set the CAT configuration for L2 cache shared by CPU0 and CPU1.
+
+a. Set CLOS with ``wrmsr <reg_num> <value>``, we want VM1 to use the lower 6 ways of cache,
+   so CLOS0 is set to 0xf0 for the upper 4 ways, and CLOS1 is set to 0x0f for the lower 4 ways::
+
+      ACRN:\>wrmsr -p1 0xd10 0xf0
+      ACRN:\>wrmsr -p1 0xd11 0x0f
+
+#. Attach COS1 to PCPU1. Because MSR IA32_PQR_ASSOC [bit 63:32], we’ll write
+   0x100000000 to it to use CLOS1::
+
+      ACRN:\>wrmsr -p0 0xc8f 0x000000000
+      ACRN:\>wrmsr -p1 0xc8f 0x100000000
+
+In addition to set the CAT configuration via HV commands, we allow developers to add
+the CAT configurations to the VM config and do the configure automatically at the
+time of RTVM creation, please refer to the :ref:`configure_cat_vm` for the details.
+
+Set up core allocation for RTVM
+-------------------------------
+In our recommended configuration, 2 cores will be allocated to RTVM: 
+core 0 for housekeeping and core 1 for RT tasks. In order to achieve 
+this, follow below steps to allocate all housekeeping tasks to core 0:
+
+.. code-block:: bash  
+
+   #!/bin/bash
+   # Move all IRQs to core 0.
+   for i in `cat /proc/interrupts | grep '^ *[0-9]*[0-9]:' | awk {'print $1'} | sed 's/:$//' `;
+   do
+       echo setting $i to affine for core zero
+       echo 1 > /proc/irq/$i/smp_affinity
+   done
+    
+   # Move all rcu tasks to core 0.
+   for i in `pgrep rcu`; do taskset -pc 0 $i; done
+    
+   # Change realtime attribute of all rcu tasks to SCHED_OTHER and priority 0
+   for i in `pgrep rcu`; do chrt -v -o -p 0 $i; done
+    
+   # Change realtime attribute of all tasks on core 1 to SCHED_OTHER and priority 0
+   for i in `pgrep /1`; do chrt -v -o -p 0 $i; done
+    
+   # Change realtime attribute of all tasks to SCHED_OTHER and priority 0
+   for i in `ps -A -o pid`; do chrt -v -o -p 0 $i; done
+
```

Comment:
Here we should disable timer migration using following command:
echo disabling timer migration
echo 0 > /proc/sys/kernel/timer_migration

Reply:
Fixed in new release.